wagtail/snippets/urls.py

```diff
@@ -11,10 +11,17 @@
     path('choose/<slug:app_label>/<slug:model_name>/', chooser.choose, name='choose'),
     path('choose/<slug:app_label>/<slug:model_name>/<str:pk>/', chooser.chosen, name='chosen'),
 
-    path('<slug:app_label>/<slug:model_name>/', snippets.list, name='list'),
+    path('<slug:app_label>/<slug:model_name>/', snippets.ListView.as_view(), name='list'),
+    path('<slug:app_label>/<slug:model_name>/results/', snippets.ListResultsView.as_view(), name='list_results'),
```

Comment:
I'm just wondering if separate views are the best way to go here? It certainly makes things quite clear-cut, but it feels like a big departure. Perhaps we could reuse the same view class, but in a different 'mode'? e.g.

```
path('<slug:app_label>/<slug:model_name>/results/', snippets.ListView.as_view(results_only=True), name='list_results'),
```

Reply:
I think the benefits of separate views will come further down the line once we split the logic out further into separate methods - the results-only view's `get_context` method can leave out a lot of the stuff that's only used for page furniture.

Reply:
Okay, but it's looking to me like pretty much everything currently happening in `get_context_data()` is relevant to both views. All that really needs to change is the template.

Reply:
OK, that's fair... I was imagining that the code for handling the search form would drop out, but (for now at least) we still need the form object in order to validate the incoming URL parameter. So yep, the `can_add_snippet` permission check is probably the only real potential saving...

So it's really just a question of whether we do the template swap with two different view classes, or by checking a `results_only` parameter in a `get_template_names` method. I don't see a huge amount of difference there, but happy to go with the latter if that will make this mergeable :-)

Reply:
I feel like the `results_only` option is 'ever-so-slightly' closer to the original solution. The concept of a view using a different template based on some value is a familiar one, and I can't think of a compelling reason to move away from it here. We're really splitting hairs by this point though :-) Go with whatever you're happy with.

Reply:
OK - updated to use a `results_only` parameter in 286ee6d412e572650a92cb0f6f4ff1083c5cfa5d.